bundles/binding/org.openhab.binding.alarmdecoder/src/main/java/org/openhab/binding/alarmdecoder/internal/ADZoneTracker.java

```diff
@@ -0,0 +1,242 @@
+
+                /** we found it so break out */
+                if( zn.getID() == _lastZone ) {
+                    bFoundLast = true;
+                }
+            }
+
+            /**
+             * Iterate through the rest till we reach the end or find our
+             * new fault adding everything we find as we go to our remove list
+             */
+            ArrayList<Integer> removeList =  new ArrayList<Integer>();
+            boolean bFoundNew = false;
+            while(litr.hasNext() && !bFoundNew) {
+                ADZone zn = (ADZone) litr.next();
+                if( zn.getID() == zone ) {
+                    bFoundNew=true;
+                    break;
+                } else {
+                    removeList.add(zn.getID());
+                    updateList.add(zn.getID());
+                }
+            }
+
+            /**
+             * start over if we didnt find the end and remove
+             * everything we find till we do reach our current fault
+             */
+             if(!bFoundNew) {
+                 litr = _zoneList.listIterator();
+                 while(litr.hasNext() && !bFoundNew) {
+                     ADZone zn = (ADZone) litr.next();
+                     if( zn.getID() == zone ) {
+                         bFoundNew=true;
+                         break;
+                     } else {
+                         removeList.add(zn.getID());
+                         updateList.add(zn.getID());
+                     }
+                 }
+             }
+
+             litr = removeList.listIterator();
+             while ( litr.hasNext() ) {
+                 Integer zid = (Integer)litr.next();
+                 ADZone zn = new ADZone(zid);
+                 _zoneList.remove(zn);
+             }
+
+         /** zone does not exist just get it into our list */
+         } else {
+
+             ADZone f = new ADZone(zone);
+
+             boolean bInserted=false;
+             int offset=0;
+             litr = _zoneList.listIterator();
+             while(litr.hasNext()) {
+                 ADZone zn = (ADZone) litr.next();
+                 if( zn.getID() > zone ) {
+                     _zoneList.add(offset, f);
+                     bInserted=true;
+                     break;
+                 }
+                 offset++;
+             }
+
+             /** just add it */
+             if ( !bInserted ) {
+                 _zoneList.add(f);
+             }
+         }
+
+         /** update our last touched zone */
+         _lastZone = zone;
+
+         setZoneState(zone,state);
+
+         return updateList;
+     }
+
+     /**
+      * Update panel Ready state clear zones if needed
+      *
+      * @parm state - the current panel READY state
+      * @return ArrayList of zones effected
+      */
+      public ArrayList updateReadyState(boolean state) {
+
+          ArrayList<Integer> updateList = new ArrayList<Integer>();
+
+          /** if we go ready clear all faults */
```

Comment:
Doesn't leading a comment with `/**` mean it [should be considered by JavaDoc](http://www.oracle.com/technetwork/java/javase/documentation/index-137868.html#terminology)?